src/HeaderCell.tsx

```diff
@@ -62,28 +71,36 @@ export default function HeaderCell<R, SR>({
     sortDirection && !priority ? (sortDirection === 'ASC' ? 'ascending' : 'descending') : undefined;
 
   const className = getCellClassname(column, column.headerCellClass, {
+    [cellSortableClassname]: column.sortable,
     [cellResizableClassname]: column.resizable
   });
 
   const renderHeaderCell = column.renderHeaderCell ?? defaultRenderHeaderCell;
 
+  function getPointerOffset(event: React.MouseEvent<HTMLDivElement>) {
+    const { right, left } = event.currentTarget.getBoundingClientRect();
+    return isRtl ? event.clientX - left : right - event.clientX;
+  }
+
   function onPointerDown(event: React.PointerEvent<HTMLDivElement>) {
     if (event.pointerType === 'mouse' && event.buttons !== 1) {
       return;
     }
 
-    const { currentTarget, pointerId } = event;
-    const { right, left } = currentTarget.getBoundingClientRect();
-    const offset = isRtl ? event.clientX - left : right - event.clientX;
+    const offset = getPointerOffset(event);
 
     if (offset > 11) {
       // +1px to account for the border size
       return;
     }
 
+    // prevent selecting/sorting if we started resizing the column, including double-clicking
+    ignoreClickRef.current = true;
```

Comment:
We do not use a ref currently. Is there a bug?

Reply:
Without it, after resizing the column, the `click` event is triggered on the header cell.
This wasn't an issue before as the `click` event handler was on a child element